environment.py

```diff
@@ -424,6 +424,11 @@ def env() -> Mapping[str, Optional[str]]:
         #
         'AZUL_TDR_SOURCES': None,
 
+        # The storage location of the BigQuery datasets for the TDR sources,
+        # e.g. `us-central1`. All sources must reside in the same region to
```

Comment:
This is going to cause issues when https://github.com/DataBiosphere/azul/pull/3441/files lands, the catalog sources for lungmap are not updated in that PR, the lungmap source lives in `US`.

```
  "name": "lungmap_dev_20210412__20210414",
  "description": "Create snapshot lungmap_dev_20210412__20210414",
  ...
  "storage": [
    {
      "region": "us",
      "cloudResource": "bigquery",
      "cloudPlatform": "gcp"
    },
...
```

Some options....
1. Ask for the Lungmap snapshots to be moved to the same location as the HCA for the DCP#17 issue.
2. Define the location within the catalog configuration, and make the reindex script purchase slots for the correct catalog.locations. This one might be expensive because the min. purchase quantity of 100.

Reply:
3. Add a flag to each catalog that controls whether slots are necessary to index that catalog. Only if the flag is set does the requirement for the configured location to match the location of all snapshots in the catalog apply.